libraries/AC_AttitudeControl/AC_PosControl_TS.cpp

```diff
@@ -0,0 +1,45 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AC_PosControl_TS.h"
+#include <AP_Math/AP_Math.h>
+#include <AP_Logger/AP_Logger.h>
+
+extern const AP_HAL::HAL& hal;
+#include "AC_PosControl_TS.h"
+
+// Default constructor.
+// Note that the Vector/Matrix constructors already implicitly zero
+// their values.
+//
+AC_PosControl_TS::AC_PosControl_TS(AP_AHRS_View& ahrs, const AP_InertialNav& inav,
+                             const AP_Motors& motors, AC_AttitudeControl& attitude_control) :
+    AC_PosControl(ahrs, inav, motors, attitude_control) 
+{
+}
+
+/// init_xy_controller - initialise the xy controller
+///     this should be called after setting the position target and the desired velocity and acceleration
+///     sets target roll angle, pitch angle and I terms based on vehicle current lean angles
+///     should be called once whenever significant changes to the position target are made
+///     this does not update the xy target.
+///     If init_I_terms is false, do not assume that current lean angles represent the desired accelerations.
+void AC_PosControl_TS::init_xy_controller(bool init_I_terms)
+{
+    // set roll, pitch lean angle targets to current attitude
+    // todo: this should probably be based on the desired attitude not the current attitude
```

Comment:
this todo would also fix? Then we could apply to all vehicles? I assume the issue is due to the large difference between pitch and desired for TVBS takeoff?

Reply:
possibly, but it would expand the scope of this change into areas I'm not familiar with and I'm not aware of any other problems that need "fixing"...

Also, that todo is from @lthall so he would probably want to specify how to implement a more general solution.
I just noticed that the commit that added that todo also removed the exact change which I've added here, with the comment "reduce glitch on init", so it may be that my change will result in glitches when switching to qloiter in flight.